base/plumbing/ServerIOBase.js

```diff
@@ -195,18 +195,18 @@ if (C.isProduction()) {
 
 ServerIO.upload = function(file, progress, load) {
 	// Provide a pre-constructed XHR so we can insert progress/load callbacks
-	const xhr = () => {
-		const request = $.ajaxSettings.xhr();
-		request.onProgress = progress;
-		request.onLoad = load; // ??@Roscoe - Any particular reason for using onLoad instead of .then? ^Dan
-		return request;
+	const makeXHR = () => {
+		const xhr = $.ajaxSettings.xhr(); //new window.XMLHttpRequest();
+		xhr.upload.addEventListener('progress', progress);
```

Comment:
Since support depends on the server -- Shall we add a toggle e.g. `ServerIO.UPLOAD_PROGRESS_SUPPORT = true;` so that the client code can switch it off?